common/header.go

```diff
@@ -5,15 +5,19 @@ package common
 import (
 	"encoding/binary"
 	"errors"
+	"sync"
 
 	"github.com/shaleman/libOpenflow/util"
 )
 
 var messageXid uint32 = 1
+var messageXidMutex sync.Mutex
 
 func NewHeaderGenerator(ver int) func() Header {
 	return func() Header {
-		messageXid += 1
+		messageXidMutex.Lock()
+		defer messageXidMutex.Unlock()
+		messageXid++
```

Comment:
This could also use `atomic.AddInt64(&messageXid, 1)` and then there's no need for a mutex at all.  Probably not worth converting the type from `int`, though